apps/bluetooth/js/modules/transfer_manager.js

```diff
@@ -0,0 +1,372 @@
+     *
+     * @access private
+     * @memberOf TransferManager
+     * @param {Object} event
+     * @param {String} event.type - type of event name
+     * @param {Object} event.detail - device info in this object
+     * @param {Object} event.detail.address - address of device
+     */
+    _onDevicePicked: function bttm__onDevicePicked(event) {
+      // TODO: send file to the device via address
+      Debug('_onDevicePicked(): address of target device = ' + 
+            event.detail.address);
+      var targetDeviceAddress = event.detail.address;
+
+      // Produce sending files schedule.
+      // Then, post message to system app.
+      this._produceSendingFilesSchedule();
+
+      // Send each file via Bluetooth sendFile API
+      var blobs = this._activity.source.data.blobs;
+      this._numberOfTasks = blobs.length;
+      blobs.forEach((blob, index) => {
+        /**
+         * Checking blob.name is because the sendFile() API needs a "file" o
+         * bject. And it is needing a filaname before send it.
+         * If there is no filename in the blob, Bluetooth API will give a 
+         * default name "Unknown.jpeg".
+         * So Bluetooth app have to find out the name via device stroage.
+         */
+        if (blob.name) {
+          // The blob has name, send the blob directly.
+          Debug('blob is sending with name...');
+          this._sendFileThenCheckTasksComplete(targetDeviceAddress, blob);
+        } else if (this._activity.source.data.filepaths) {
+          // The blob does not have name,
+          // browse the file via filepath from storage again.
+          var filepath = this._activity.source.data.filepaths[index];
+          var storage = navigator.getDeviceStorage('sdcard');
+          var getRequest = storage.get(filepath);
+          getRequest.onsuccess = () => {
+            Debug('getFile succeed & file is sending...');
+            this._sendFileThenCheckTasksComplete(targetDeviceAddress, 
+                                                 getRequest.result);
+          };
+
+          getRequest.onerror = () => {
+            Debug('getFile failed so that blob is sending without filename ' +
+                  getRequest.error && getRequest.error.name);
+            this._sendFileThenCheckTasksComplete(targetDeviceAddress, blob);
+          };
+        } else {
+          // The blob does not have name and filepath,
+          // pass the blob without name to send file directly.
+          Debug('no filepath to get from device storage ' +
+                'so that blob is sending without filename');
+          this._sendFileThenCheckTasksComplete(targetDeviceAddress, blob);
+        }
+      });
+    },
+
+    /**
+     * Given address, blob(file) to send file via platform.
+     *
+     * @access private
+     * @memberOf TransferManager
+     * @param {String} address - address of target device
+     * @param {Object} blob - blob(file) to send
+     */
+    _sendFileThenCheckTasksComplete: 
```

Comment:
`_sendFileThenCheckTasksComplete` and `_checkTasksComplete` can be removed if we are using Promise.all.